### PR TITLE
REALMC-8271: rebase off current goja to support trimLeft and trimRight

### PR DESCRIPTION
**Pull request by tsedgwick**

### CODE REVIEW

**Review by arahmanan — COMMENTED** [review]
@tsedgwick I believe the forked repo has recently added support for `trimRight` and `trimLeft`. Should we pull that down instead?

**Comment by tsedgwick** [comment]
sorry @arahmanan - it went from a small PR to a large one with your good suggestion 😎

**Comment by Gabri3l** [comment]
Quick question, this ended up just being a rebase with no additional changes from our side ?

**Comment by tsedgwick** [comment]
> Quick question, did this end up being a rebase with no additional changes from us ?

Yes, this was for trimLeft and trimRight - both of which the rebase takes care of.

File: builtin_string.go [hunk 1]
```diff
@@ -951,8 +988,12 @@ func (r *Runtime) initString() {
...
 	o._putProp("toUpperCase", r.newNativeFunc(r.stringproto_toUpperCase, nil, "toUpperCase", nil, 0), true, false, true)
 	o._putProp("trim", r.newNativeFunc(r.stringproto_trim, nil, "trim", nil, 0), true, false, true)
-	o._putProp("trimEnd", r.newNativeFunc(r.stringproto_trimEnd, nil, "trimEnd", nil, 0), true, false, true)
-	o._putProp("trimStart", r.newNativeFunc(r.stringproto_trimStart, nil, "trimStart", nil, 0), true, false, true)
+	trimEnd := r.newNativeFunc(r.stringproto_trimEnd, nil, "trimEnd", nil, 0)
+	trimStart := r.newNativeFunc(r.stringproto_trimStart, nil, "trimStart", nil, 0)
+	o._putProp("trimEnd", trimEnd, true, false, true)
+	o._putProp("trimStart", trimStart, true, false, true)
+	o._putProp("trimRight", trimEnd, true, false, true)
+	o._putProp("trimLeft", trimStart, true, false, true)
```
Comment (Gabri3l) [inline builtin_string.go:996]: TIL `trimLeft` is just an alias for `trimStart`, although I never used `trimStart` before. Double TIL

**Review by arahmanan — COMMENTED** [review]
Half way through!

**Review by arahmanan — APPROVED** [review]
LGTM! Pending the minor comments above. I

**Pull request merged**